internal/brokers/examplebroker.go

```diff
@@ -85,9 +85,24 @@ func newExampleBroker(name string) (b *exampleBroker, fullName, brandIcon string
 	}, strings.ReplaceAll(name, "_", " "), fmt.Sprintf("/usr/share/brokers/%s.png", name)
 }
 
-// GetAuthenticationModes returns the list of supported authentication modes for the selected broker depending on user name.
-func (b *exampleBroker) GetAuthenticationModes(ctx context.Context, username, lang string, supportedUILayouts []map[string]string) (sessionID, encryptionKey string, authenticationModes []map[string]string, err error) {
+// NewSession creates a new session for the specified user.
+func (b *exampleBroker) NewSession(ctx context.Context, username, lang string) (sessionID, encryptionKey string, err error) {
 	sessionID = uuid.New().String()
+	b.currentSessionsMu.Lock()
+	b.currentSessions[sessionID] = sessionInfo{
+		username: username,
+		lang:     lang,
+	}
+	b.currentSessionsMu.Unlock()
+	return sessionID, brokerEncryptionKey, nil
+}
+
+// GetAuthenticationModes returns the list of supported authentication modes for the selected broker depending on session info.
+func (b *exampleBroker) GetAuthenticationModes(ctx context.Context, sessionID string, supportedUILayouts []map[string]string) (authenticationModes []map[string]string, err error) {
+	session, ok := b.currentSessions[sessionID]
```

Comment:
Shouldn’t that use the mutex?

Reply:
I'm not sure, to be honest. Since it's just reading, it should be fine, no?